prompt/prompt-executor/prompt-executor-clients/prompt-executor-ollama-client/src/commonMain/kotlin/ai/koog/prompt/executor/ollama/client/OllamaModelManager.kt

```diff
@@ -0,0 +1,205 @@
+import io.ktor.client.request.*
+import io.ktor.http.*
+import kotlin.time.Clock
+import kotlin.time.ExperimentalTime
+
+/**
+ * Manages Ollama models, providing dynamic discovery and capabilities detection.
+ */
+public class OllamaModelManager(
+    private val client: HttpClient,
+    private val baseUrl: String
+) {
+    private val logger = KotlinLogging.logger { }
+
+    /**
+     * Cache for discovered models to avoid repeated API calls.
+     */
+    private var modelsCache: List<OllamaModelInfo>? = null
+    private var lastCacheUpdate: Long = 0
+    private val cacheValidityMs = 300_000L // 5 minutes
+
+    /**
+     * Dynamic capability detection based on model family - minimal hardcoding.
+     * We only define the general capability patterns, not specific models.
+     */
+    private fun getCapabilitiesForFamily(family: String): List<LLMCapability> {
+        return when (family) {
+            "embedding" -> listOf(LLMCapability.Embed)
+            "llama", "qwen", "gemma", "mistral", "deepseek" -> listOf(
+                LLMCapability.Temperature,
+                LLMCapability.Schema.JSON.Simple,
+                LLMCapability.Tools,
+                LLMCapability.Completion
+            )
+            "phi", "codellama" -> listOf(
+                LLMCapability.Temperature,
+                LLMCapability.Completion
+            )
+            else -> listOf(
+                LLMCapability.Temperature,
+                LLMCapability.Completion
+            )
+        }
+    }
+
+    /**
+     * Retrieves all available models from Ollama.
+     */
+    @OptIn(ExperimentalTime::class)
+    public suspend fun getAvailableModels(): List<OllamaModelInfo> {
+        val now = Clock.System.now().epochSeconds
+        
+        // Return cached models if still valid
+        if (modelsCache != null && (now - lastCacheUpdate) < cacheValidityMs) {
+            return modelsCache!!
+        }
+
+        return try {
+            val response = client.get("$baseUrl/api/tags") {
+                contentType(ContentType.Application.Json)
+            }.body<OllamaModelsListResponse>()
+            
+            modelsCache = response.models
+            lastCacheUpdate = now
+            
+            logger.info { "Discovered ${response.models.size} Ollama models" }
+            response.models
+        } catch (e: Exception) {
+            logger.error(e) { "Failed to fetch available models from Ollama" }
+            emptyList()
+        }
+    }
+
+    /**
+     * Resolves a model name to an LLModel instance with detected capabilities.
+     */
+    public suspend fun resolveModel(modelName: String): LLModel? {
+        val availableModels = getAvailableModels()
+        val matchedModel = findModelByName(availableModels, modelName)
+            ?: return null
+
+        val capabilities = detectCapabilities(matchedModel)
+        
+        return LLModel(
+            provider = LLMProvider.Meta, // Default to Meta for now
```

Comment:
You should create a `LLMProvider.Ollama` provider.

The LLMProvider seems to represent who provides the model, not who did the model. For example, `OpenRouter` is an LLMProvider, though it provides models made by Meta, Microsoft, etc.

Also, we need to change the LLModels defined in `OllamaModels` to use that new provider.